README.md

```diff
@@ -33,309 +33,181 @@ Using RealtimeSanitizer requires RADSan-enabled `clang` and two actions:
 
 # Usage
 
-See [Example Usage](#example-usage).
+Using RTSan is very simple. When compiling with LLVM 20, only two steps are required:
 
+1. mark your real-time algorithm entrypoint function with the `[[clang::nonblocking]]` attribute, and
+2. add `-fsanitize=realtime` to your compile and link flags.
 
 ## CMake
 
-RADSan-enabled clang is installed to `/usr/local` in the RADSan Docker image,
-and CMake will automatically detect it. However, if you've built RADSan from
-source, you'll need to instruct CMake to use it by either:
+LLVM 20 `clang` is installed to `/usr/local` in the RTSan Docker image, and
+CMake will automatically detect it. However, if you've built RTSan from source,
+you'll need to instruct CMake to use it by either setting the `CC` and `CXX`
+environment variables or passing the `CMAKE_C_COMPILER` and
+`CMAKE_CXX_COMPILER` options to CMake as arguments.
 
-1. setting the `CC` and `CXX` environment variables,
+Adding the compile and link flag `-fsanitize=realtime` can be done however
+works best for your project. One unintrusive option is to pass them as
+arguments to CMake:
 
 ```sh
-CC=/path/to/built/clang CXX=/path/to/built/clang++ cmake ..
-```
-
-2. passing the `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` options to cmake
-
-```sh
-cmake -DCMAKE_C_COMPILER=/path/to/built/clang -DCMAKE_CXX_COMPILER=/path/to/built/clang++
-```
-
-3. or using `set(CMAKE_CXX_COMPILER ...)` in your CMake project file.
-
-
-```cmake
-set(CMAKE_CXX_COMPILER /path/to/built/clang++)
-set(CMAKE_C_COMPILER   /path/to/built/clang)
+cmake \
+  -DCMAKE_CXX_FLAGS="-fsanitize=realtime" \
+  -DCMAKE_C_FLAGS="-fsanitize=realtime" \
+  -DCMAKE_EXE_LINKER_FLAGS="-fsanitize=realtime" ...
 ```
 
-Adding the compiler flag `-fsanitize=realtime` can be done however works best
-for your project. Individual targets can be configured with:
-
-```cmake
-target_compile_options(MyTarget PUBLIC -fsanitize=realtime)
-target_link_options(MyTarget PUBLIC -fsanitize=realtime)
-```
+# How it works
 
-or add the following to configure all targets in the project:
+RTSan's algorithm consists of two parts that work together:
 
-```cmake
-add_compile_options(-fsanitize=realtime)
-add_link_options(-fsanitize=realtime)
-```
+- A compilation step that
+  - signals real-time context entry and exit for the duration of `[[clang::nonblocking]]` functions, and
+- A lightweight runtime library that
+  - keeps track of the real-time context,
+  - detects calls to system library functions that are known to block, and
+  - raises an error if a blocking library function is called in a real-time context.
 
 # Configuration
 
-## Error modes
-
-RADSan's behaviour during real-time violations can be configured using the `RADSAN_ERROR_MODE`
-environment variable. RADSan recognises any of the following values for `RADSAN_ERROR_MODE`:
+## Disabling RTSan
```

Comment:
Another good candidate for cutting (as I have that PR to add it to the upstream docs)